### PR TITLE
Updates to Sheets API index

### PR DESCRIPTION
**Pull request by donaldrgosselin**
image
image
image
image
image
image

### CODE REVIEW

File: api/sheets/get_name.md [hunk 1]
```diff
@@ -6,41 +6,60 @@ parent: Google Sheets
 grand_parent: API
 ---
 
-# Get a node's name
+# Returning the Names Associated with a DCID
```
Comment (beets) [inline api/sheets/get_name.md:9]:
```suggestion
# Returning the Names Associated with a Node
```

File: api/sheets/get_name.md [hunk 2]
```diff
@@ -6,41 +6,60 @@ parent: Google Sheets
...
 
-Given a cell or a column range of cells with [DCIDs](/glossary.html#dcid) (unique identifiers for all entities in the Data Commons knowledge graph), return the names associated with the given DCIDs.
+The `=DCGETNAME(*dcids*)` formula returns the names associated with given DCIDs to a cell or a column range of cells.
```
Comment (beets) [inline api/sheets/get_name.md:11]:
```suggestion
The `=DCGETNAME(*dcids*)` formula returns the names associated with given node DCIDs to a cell or a column range of cells.
```

File: api/sheets/get_name.md [hunk 3]
```diff
@@ -6,41 +6,60 @@ parent: Google Sheets
...
+> **NOTE:**
+> 
+> Be sure to follow the instructions for [Installing and Enabling the Sheets Add-On](https://docs.datacommons.org/api/sheets/) before using this formula.
 
-**Formula**: `=DCGETNAME(dcids)`
+## Formula {#formula}
 
-**Required arguments**:
+```
+=DCGETNAME(*dcids*)
```
Comment (beets) [inline api/sheets/get_name.md:20]:
```suggestion
=DCGETNAME(*DCIDs*)
```

File: api/sheets/get_name.md [hunk 4]
```diff
@@ -6,41 +6,60 @@ parent: Google Sheets
...
+```
 
-* `dcids` - A list of nodes to query, identified by their DCIDs.
+## Required Arguments {#required-arguments}
 
-## Examples
+`dcids` - A list of DCID nodes to query
```
Comment (beets) [inline api/sheets/get_name.md:25]:
```suggestion
`DCIDs` - A single or list of node DCIDs to query
```

File: api/sheets/get_name.md [hunk 5]
```diff
@@ -6,41 +6,60 @@ parent: Google Sheets
...
 
-Before trying this method out, make sure to follow the setup directions in [the main section for Sheets docs](/api/sheets/index.html).
+## Returns {#returns}
 
-### Example 1: Retrieve the name of a country by its DCID.
+The formula returns the name associated with a specified DCID. See the [Examples](#examples) section below for examples of positive returns. If a DCID does not exist, the `=DCGETNAME(*dcids*)` formula does not return a value. For example, because the “geoId/123123123” DCID does not exist, no value is returned to cell B1 in the following sheet:
```
Comment (beets) [inline api/sheets/get_name.md:29]:
```suggestion
The formula returns the name associated with a specified DCID. See the [Examples](#examples) section below for examples of positive return values.

If a DCID does not exist, the `=DCGETNAME(*dcids*)` formula does not return a value. For example, because the “geoId/123123123” DCID does not exist, no value is returned to cell B1 in the following sheet:
```

File: api/sheets/get_name.md [hunk 6]
```diff
@@ -6,41 +6,60 @@ parent: Google Sheets
...
 
-![](/assets/images/sheets/sheets_get_name_cote_d_ivoire_cropped.png)
+![alt_text](images/sheets_get_return_error_1.png "image_tooltip")
```
Comment (beets) [inline api/sheets/get_name.md:31]: Please provide better tooltip text

File: api/sheets/get_name.md [hunk 7]
```diff
@@ -6,41 +6,60 @@ parent: Google Sheets
...
 
-### Example 2: Retrieve the names of a collection of cell lines.
+If you provide an empty cell for a DCID, the `=DCGETNAME(*dcids*)` formula returns a value of #ERROR!, as shown show in the following image:
 
-![](/assets/images/sheets/sheets_get_name_cell_lines_cropped.png)
+![alt_text](images/sheets_get_return_error_2.png "image_tooltip")
```
Comment (beets) [inline api/sheets/get_name.md:35]: Please provide better tooltip text

File: api/sheets/get_name.md [hunk 8]
```diff
@@ -6,41 +6,60 @@ parent: Google Sheets
...
 
->  **TIP:**
->
->  It's best to minimize the number of function calls to `DCGETNAME` by using a single call to get names for a column of nodes. This is because a spreadsheet will make one call to a Google server [per custom function call](https://developers.google.com/apps-script/guides/sheets/functions#optimization). If your sheet contains thousands of separate calls to `DCGETNAME`, expect it to be slow and with errors.
+## Examples {#examples}
 
+This section contains examples of returning the names associated with given DCIDs.
```
Comment (beets) [inline api/sheets/get_name.md:39]:
```suggestion
This section contains examples of returning the names associated with given node DCIDs.
```

File: api/sheets/get_name.md [hunk 9]
```diff
@@ -6,41 +6,60 @@ parent: Google Sheets
...
+
+### Example 2: Retrieve the Names of a Collection of Cells
+
+To retrieve the names of a collection of cells using DCIDs, perform the following steps:
+
+1. Using the following image, enter the DCIDs for four African nations into column A.
+2. Place your cursor in cell B1 and type the following formula: **`=DCGETNAME`**.
+3. Press **Enter** and the names of the four African nations populates column B.
+
+![alt_text](images/sheets_get_name_example_2.png "image_tooltip")
```
Comment (beets) [inline api/sheets/get_name.md:60]: Please give better tooltip text

File: api/sheets/get_name.md [hunk 10]
```diff
@@ -6,41 +6,60 @@ parent: Google Sheets
...
+
+> **IMPORTANT**
+> 
+> It’s best to minimize the number of function calls to `=DCGETNAME(*dcids*)` by using a single call to get the names for a column of nodes. This is because a spreadsheet will make one call to a Google server _[per custom function call](https://developers.google.com/apps-script/guides/sheets/functions#optimization)_. If your sheet contains many separate calls to `=DCGETNAME(*dcids*)`, you can expect it to be slow and return errors.
```
Comment (beets) [inline api/sheets/get_name.md:64]:
```suggestion
> **IMPORTANT**
> 
> It’s best to minimize the number of function calls to `=DCGETNAME(*dcids*)` by using a single call to get the names for a column of nodes. This is because tyhe spreadsheet will make one call to a Google server _[per custom function call](https://developers.google.com/apps-script/guides/sheets/functions#optimization)_. If your sheet contains many separate calls to `=DCGETNAME(*dcids*)`, you can expect it to be slow and return errors.
```

File: api/sheets/index.md [hunk 11]
```diff
@@ -8,50 +8,75 @@ has_children: true
...
 Also read our [step-by-step guides](tutorials/) for more examples.
 
-## Getting started
+## Installing and Enabling the Sheets Add-On
 
-Install the add-on from [G Suite Marketplace](https://gsuite.google.com/marketplace/app/data_commons/454343067575). To enable the add-on in a document, click "Add-ons > Data Commons > Fill place dcids". You may use the resulting sidebar to start [finding dcids](/api/sheets/get_dcid.html) in the United States, or close it and reopen it at any time. Note that none of the custom functions will work in a given document until you have enabled the add-on by choosing "Fill place dcids".
+Install and enable the Sheets add-on from Google Workspace Marketplace, as follows:
 
-![](/assets/images/sheets/sheets_menu_bar.png)
+1. Install the Sheets add-on by clicking **[Google Workspace Marketplace](https://gsuite.google.com/marketplace/app/data_commons/454343067575)** (https://gsuite.google.com/marketplace/app/data_commons/454343067575).
```
Comment (beets) [inline api/sheets/index.md:23]:
```suggestion
1. Install the Sheets add-on by clicking **[Google Workspace Marketplace](https://gsuite.google.com/marketplace/app/data_commons/454343067575)** .
```

File: api/sheets/index.md [hunk 12]
```diff
@@ -8,50 +8,75 @@ has_children: true
...
 
-Data Commons' Sheets add-on ships with the ability to look up a location's [DCID](/glossary.html), or unique Data Commons identifier, within the application. To use this feature, select the final destination cell of the place [DCID](/glossary.html). Go to the Sheets menu bar, and click:
 
-**"Add-ons > Data Commons > Fill place dcids"**
 
-![](/assets/images/sheets/sheets_menu_bar.png)
+## Finding a Place’s DCID
 
-A sidebar will appear on the right of the sheet, with a search bar where you can start typing the name of the place you want. From the drop down menu, select the place you want, and its [DCID](/glossary.html) will appear in the cell that you selected.
+Data Commons’ Sheets add-on ships with the ability to look up a location’s [DCID](https://docs.datacommons.org/glossary.html), within the application by using the **Fill place dcids** feature. To use this feature to find a place’s DCID:
```
Comment (beets) [inline api/sheets/index.md:34]:
```suggestion
Data Commons’ Sheets add-on provides the ability to look up a location’s [DCID](https://docs.datacommons.org/glossary.html), within the application by using the **Fill place dcids** feature. To use this feature to find a place’s DCID:
```

File: api/sheets/index.md [hunk 13]
```diff
@@ -8,50 +8,75 @@ has_children: true
...
 Note that this feature only supports places in the United States.
 
-## Sorting your results
+## Introduction to the Sheets API Custom Functions
+
+The Data Commons Sheets API includes the five formulas listed in the following table. Click the links in the table for detailed information on each formula.
+
+| **Formula**                                                                                  | **Description**                           |
+|----------------------------------------------------------------------------------------------|-------------------------------------------|
+| [=DCGETNAME(_dcids_)](https://docs.datacommons.org/api/sheets/get_name.html)                 | Returns the names associated with a DCID. |
```
Comment (beets) [inline api/sheets/index.md:49]:
```suggestion
| [=DCGETNAME(_DCIDs_)](https://docs.datacommons.org/api/sheets/get_name.html)                 | Returns the names associated with a DCID. |
```

Comment (beets) [reply api/sheets/index.md:49]: Here and below

File: api/sheets/index.md [hunk 14]
```diff
@@ -8,50 +8,75 @@ has_children: true
...
+| [=DCPROPERTY(_dcids_)](https://docs.datacommons.org/api/sheets/get_property.html)            | Retrieves node property values.           |
+| [=DCCOHORTMEMBERS(_dcids_)](https://docs.datacommons.org/api/sheets/get_cohort_members.html) | Gets the cohort members of a node.        |
+
+See the links in the preceding table to subsequent pages in this section for detailed descriptions of each API.
 
-<p>To sort columns of data from the API, we suggest using the "filter views" Google Sheet feature.
+## Sorting your Results
 
-<p>Please check the following video for a complete example:</p>
+To sort columns of data from the API, consider using Google Sheets “filter views” feature, which allows you to alphabetically and numerically sort and filter data
```
Comment (beets) [inline api/sheets/index.md:59]:
```suggestion
To sort columns of data from the API, use the Google Sheets “filter views” feature, which allows you to alphabetically and numerically sort and filter data
```

**Review by beets — COMMENTED** [review]
Thanks for the PR. I've added mostly suggested changes, and some comments.

When we replace images, could we please replace them with the same filename. Then we can see the difference from old to new. Please do not use _dg_ in the filenames to differentiate.

File: api/sheets/get_name.md [hunk 15]
```diff
@@ -6,41 +6,62 @@ parent: Google Sheets
...
+
+1. Using the following image, enter the DCIDs for four African nations into column A.
+2. Place your cursor in cell B1 and type the following formula: **`=DCGETNAME`**.
+3. Press **Enter** and the names of the four African nations populates column B.
+
+![alt_text](images/sheets_get_name_example_2.png "image_tooltip")
+
+> **IMPORTANT**
+> 
+> It’s best to minimize the number of function calls to `=DCGETNAME(*dcids*)` by using a single call to get the names for a column of nodes. This is because tyhe spreadsheet will make one call to a Google server _[per custom function call](https://developers.google.com/apps-script/guides/sheets/functions#optimization)_. If your sheet contains many separate calls to `=DCGETNAME(*dcids*)`, you can expect it to be slow and return errors.
```
Comment (beets) [inline api/sheets/get_name.md:66]:
```suggestion
> It’s best to minimize the number of function calls to `=DCGETNAME(*dcids*)` by using a single call to get the names for a column of nodes. This is because the spreadsheet will make one call to a Google server _[per custom function call](https://developers.google.com/apps-script/guides/sheets/functions#optimization)_. If your sheet contains many separate calls to `=DCGETNAME(*dcids*)`, you can expect it to be slow and return errors.
```

**Comment by beets** [comment]
Hi Don - I've added most of my suggested changes to the PR, so please do `git pull origin` in your branch before continuing with local changes to resolve the other outstanding comments. Thanks!

**Review by donaldrgosselin — COMMENTED** [review]
Excellent comments Carolyn. Thank you so much.

**Comment by donaldrgosselin** [comment]
> Thanks for the PR. I've added mostly suggested changes, and some comments.
> 
> When we replace images, could we please replace them with the same filename. Then we can see the difference from old to new. Please do not use _dg_ in the filenames to differentiate.

Great suggestions, thanks Carolyn. Sorry about the "dg" filename issue.

File: Gemfile [hunk 16]
```diff
@@ -8,6 +8,7 @@ source "https://rubygems.org"
 #
 # To upgrade, run `bundle update github-pages`.
 gem "github-pages", group: :jekyll_plugins
+gem "webrick"
```
Comment (beets) [inline Gemfile:11]: What is this new package for? Can you please revert this change.

File: Gemfile.lock [hunk 17]
```diff
@@ -1,61 +1,49 @@
 GEM
   remote: https://rubygems.org/
   specs:
-    activesupport (6.0.6.1)
+    activesupport (7.0.8)
```
Comment (beets) [inline Gemfile.lock:4]: Please remove this file's updates from the PR

File: api/sheets/get_name.md [hunk 18]
```diff
@@ -24,23 +24,23 @@ Before trying this method out, make sure to follow the setup directions in [the
 
 ### Example 1: Retrieve the name of a country by its DCID.
 
-![](/assets/images/sheets/sheets_get_name_cote_d_ivoire_cropped.png)
+![](../../assets/images/sheets/sheets_get_name_cote_d_ivoire_cropped.png)
```
Comment (beets) [inline api/sheets/get_name.md:27]: As mentioned before, please replace the old images, rather than add new filenames (like _cropped).

Also, the file paths should start with /assets/images, not ../../

This comment applies to all other images in these PRs

File: api/sheets/get_name.md [hunk 19]
```diff
@@ -24,23 +24,23 @@ Before trying this method out, make sure to follow the setup directions in [the
```
Comment (beets) [inline api/sheets/get_name.md:24]: If there's another outstanding PR for this file, can we remove it from this PR please?

File: api/sheets/index.md [hunk 20]
```diff
@@ -8,38 +8,58 @@ has_children: true
...
+## Installing and Enabling the Sheets Add-On
 
-![](/assets/images/sheets/sheets_menu_bar.png)
+Install and enable the Sheets add-on from Google Workspace Marketplace, as follows:
 
-## Finding a place's DCID
+1. Install the Sheets add-on by clicking **[Google Workspace Marketplace](https://gsuite.google.com/marketplace/app/data_commons/454343067575)** .
+2. Enable the Sheets add-on by clicking the Google Sheets **Extensions** menu, pointing to **Data Commons**, and then by clicking **Fill place dcids**.
+3. Use the resulting sidebar that appears on the right side of the page to start [finding DCIDs](https://docs.datacommons.org/api/sheets/get_name.html) in the United States. You can close and reopen the sidebar at any time.
```
Comment (beets) [inline api/sheets/index.md:25]: please use a relative path (i.e. drop https://docs.datacommons.org from the URL). this comment applies to all URLs in these PRs

File: api/sheets/index.md (hunk 13 above)
Comment (beets) [reply api/sheets/index.md:50]: This change hasn't been made.

File: api/sheets/index.md (hunk 14 above)
Comment (beets) [reply api/sheets/index.md:60]: This change hasn't been made

Comment (beets) [reply api/sheets/index.md:60]: Drop the word "consider" since this is the only way to sort.

**Review by beets — COMMENTED** [review]
We can't review your updates unless you do a `git push origin` on this branch. I've unresolved comments because they haven't been applied yet to the changes that are currently showing up in this PR.

Please remove all changes from this PR that doesn't apply to the index.md files
As mentioned before, please also remove the green borders around the screenshots.

File: api/sheets/index.md (hunk 13 above)
Comment (donaldrgosselin) [reply api/sheets/index.md:50]: I made all the dcids lowercase and removed the underscore characters to make them consistent with the rest of the pages.

File: api/sheets/index.md (hunk 14 above)
Comment (donaldrgosselin) [reply api/sheets/index.md:60]: I fixed it.

File: api/sheets/index.md (hunk 20 above)
Comment (donaldrgosselin) [reply api/sheets/index.md:25]: Ok

File: api/sheets/get_name.md (hunk 18 above)
Comment (donaldrgosselin) [reply api/sheets/get_name.md:27]: Ok

File: Gemfile (hunk 16 above)
Comment (donaldrgosselin) [reply Gemfile:11]: The package is necessary for me to run Jekyll, but I'll find another workaround.

**Comment by beets** [comment]
thanks for the updates. please do not delete get_name.md and Gemfile.lock -- just restore them to the versions at HEAD.

File: api/sheets/index.md [hunk 21]
```diff
@@ -8,38 +8,58 @@ has_children: true
...
 
-Data Commons' Sheets add-on ships with the ability to look up a location's [DCID](/glossary.html), or unique Data Commons identifier, within the application. To use this feature, select the final destination cell of the place [DCID](/glossary.html). Go to the Sheets menu bar, and click:
+## Finding a Place’s DCID
 
-**"Add-ons > Data Commons > Fill place dcids"**
+Data Commons’ Sheets add-on provides the ability to look up a location’s [DCID](/glossary.html), within the application by using the **Fill place dcids** feature. To use this feature to find a place’s DCID:
 
-![](/assets/images/sheets/sheets_menu_bar.png)
+1. Select the destination cell where you want to add a place’s [DCID](/glossary.html).
+2. Click in the search box for the **Fill place dcids for selected cells** section.
```
Comment (beets) [inline api/sheets/index.md:38]: Can we be more explicit about where this box is, like in the former text?

**Review by beets — COMMENTED** [review]
One final comment:

Please update the screenshot so that: assets/images/sheets/sheets_menu_bar.png
- it's less than 800px wide
- it doesn't include the "fill place dcids" panel on the right

**Comment by donaldrgosselin** [comment]
> One final comment:
> 
> Please update the screenshot so that: assets/images/sheets/sheets_menu_bar.png
> 
> * it's less than 800px wide
> * it doesn't include the "fill place dcids" panel on the right

I've resized both images on the page so they are 800px in width.

File: api/sheets/index.md (hunk 21 above)
Comment (donaldrgosselin) [reply api/sheets/index.md:38]: I added some additions to the text. Please let me know what you think.

**Comment by beets** [comment]
Sorry I realized my last comment was unclear.

The image (assets/images/sheets/sheets_menu_bar.png ) should not include the "fill place dcid" sidebar since we are just installing / enabling the plugin. Could you please adjust the image?

**Review by beets — COMMENTED** [review]
@shifucun I've made a few updates - ptal

**Comment by beets** [comment]
[Current screenshot](https://github.com/datacommonsorg/docsite/assets/6052978/254b1f83-93fb-4f34-b99a-43d1b7aa177d)

File: api/sheets/index.md [hunk 22]
```diff
@@ -8,38 +8,58 @@ has_children: true
...
-Install the add-on from [G Suite Marketplace](https://gsuite.google.com/marketplace/app/data_commons/454343067575). To enable the add-on in a document, click "Add-ons > Data Commons > Fill place dcids". You may use the resulting sidebar to start [finding dcids](/api/sheets/get_dcid.html) in the United States, or close it and reopen it at any time. Note that none of the custom functions will work in a given document until you have enabled the add-on by choosing "Fill place dcids".
+> **NOTE:**
+>
+> None of the custom Data Commons Sheets functions (described below) will work in a given document until you have enabled the add-on by opening the **Fill place dcids** sidebar.
 
-![](/assets/images/sheets/sheets_menu_bar.png)
+## Finding a Place’s DCID
 
-## Finding a place's DCID
+Data Commons’ Sheets add-on provides the ability to look up a location’s [DCID](/glossary.html#dcid), within the application by using the **Fill place dcids** feature. To use this feature to find a place’s DCID:
```
Comment (shifucun) [inline api/sheets/index.md:35]: should we replace "location" with "place"?

File: api/sheets/index.md [hunk 23]
```diff
@@ -8,38 +8,58 @@ has_children: true
...
-![](/assets/images/sheets/sheets_menu_bar.png)
+## Introduction to the Sheets API Custom Functions
 
-A sidebar will appear on the right of the sheet, with a search bar where you can start typing the name of the place you want. From the drop down menu, select the place you want, and its [DCID](/glossary.html) will appear in the cell that you selected.
+The Data Commons Sheets API includes the five formulas listed in the following table. Click the links in the table for detailed information on each formula.
 
-![](/assets/images/sheets/sheets_search_box.png)
+| **Formula**                                                                                  | **Description**                           |
+|----------------------------------------------------------------------------------------------|-------------------------------------------|
+| [=DCGETNAME(dcids)](/api/sheets/get_name.html)                 | Returns the names associated with a DCID. |
```
Comment (shifucun) [inline api/sheets/index.md:50]: Use "Retrieves" instead of "Returns" throughout?

**Review by shifucun — APPROVED** [review]
Thanks for completing this PR!

File: api/sheets/index.md (hunk 22 above)
Comment (beets) [reply api/sheets/index.md:35]: good catch - done

File: api/sheets/index.md (hunk 23 above)
Comment (beets) [reply api/sheets/index.md:50]: good call - done

**Review by beets — COMMENTED** [review]
thanks for the review!

File: api/sheets/index.md [hunk 24]
```diff
@@ -8,50 +8,75 @@ has_children: true
 
 # Data Commons Sheets API
 
-The **Data Commons Sheets API** is a Google Sheets add-on that enables Google Sheets
-users to import data from the Data Commons knowledge graph. The add-on provides an
-interface for finding a location's [DCID](/glossary.html), as well as some custom functions for
-importing data to a spreadsheet.
+The **Data Commons Sheets API** is a Google Sheets add-on that enables Google Sheets users to import data from the Data Commons knowledge graph. The add-on provides an interface for finding a location’s unique Data Commons identifier ([DCID](glossary.html), and some custom functions for importing data into a spreadsheet.
```
[the remark this answers is not in the document]
Comment (beets) [reply api/sheets/index.md]: done
